game/state/battle/battle.cpp

```diff
@@ -3366,7 +3366,7 @@ void Battle::exitBattle(GameState &state)
 	}
 
 	// Give player vehicle a null cargo just so it comes back to base once
-	for (auto v : returningVehicles)
+	for (auto v : playerVehicles)
```

Comment:
If this is a different issue maybe a different PR is useful?

Keeping one-logical-chance-per-PR often helps later for reading logs or bisecting

Reply:
This was wrongly copied from https://github.com/OpenApoc/OpenApoc/pull/1379

Reply:
Done, this now is reverted.